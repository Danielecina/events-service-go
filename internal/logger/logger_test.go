package logger

import (
	"bytes"
	"log"
	"testing"

	snaps "github.com/gkampitakis/go-snaps/snaps"
)

func TestInfo(t *testing.T) {
	var buf bytes.Buffer
	AppLogger = log.New(&buf, "[APP] 2006/01/02 15:04:05 logger.go:12: ", 0)
	Info("test info %s", "message")
	output := buf.String()
	snaps.MatchSnapshot(t, output)
}

func TestError(t *testing.T) {
	var buf bytes.Buffer
	AppLogger = log.New(&buf, "[APP] 2006/01/02 15:04:05 logger.go:19: ", 0)
	Error("test error %d", 123)
	output := buf.String()
	snaps.MatchSnapshot(t, output)
}

func TestDebug(t *testing.T) {
	var buf bytes.Buffer
	AppLogger = log.New(&buf, "[APP] 2006/01/02 15:04:05 logger.go:26: ", 0)
	Debug("test debug %v", true)
	output := buf.String()
	snaps.MatchSnapshot(t, output)
}
