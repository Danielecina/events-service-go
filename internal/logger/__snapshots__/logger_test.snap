
[TestInfo - 1]
[APP] 2006/01/02 15:04:05 logger.go:12: [INFO] test info message

---

[TestError - 1]
[APP] 2006/01/02 15:04:05 logger.go:19: [ERROR] test error 123

---

[TestDebug - 1]
[APP] 2006/01/02 15:04:05 logger.go:26: [DEBUG] test debug true

---
